grpclb/src/main/java/io/grpc/grpclb/GrpclbClientLoadRecorder.java

```diff
@@ -38,8 +40,9 @@
   private final AtomicLong callsFinished = new AtomicLong();
 
   // Specific finish types
-  private final AtomicLong callsDroppedForRateLimiting = new AtomicLong();
-  private final AtomicLong callsDroppedForLoadBalancing = new AtomicLong();
+  // Access to it should be protected by lock.  Contention is not an issue for these counts, because
+  // normally only a small portion of all RPCs are dropped.
+  private HashMap<String, AtomicLong> callsDroppedPerToken = new HashMap<String, AtomicLong>();
```

Comment:
I use AtomicLong for its mutability. If I used Long, each update would be two map operations, and one or more object allocation, which is presumably more expensive than incrementing AtomicLong.